lib/monitoring/pub_sub.rb

```diff
@@ -0,0 +1,33 @@
+require 'singleton'
+require 'active_support/notifications'
+
+module Monitoring
+  # PubSub and PubSubBase wrap ActiveSupport::Notifications, providing pub/sub
+  # plumbing to custom controllers and collectors.
+
+  class PubSubBase
+    attr_reader :options
+
+    def initialize(notifications = ActiveSupport::Notifications)
```

Comment:
I'm thinking this dependency injection of `notifications` might not be necessary since this particular version of pub/sub is fundamentally tied to `ActiveSupport::Notifications`. There's no other notifications object that will have the method signatures for `instrument`, `subscribe` etc.